### PR TITLE
cody 5.5.15 (new formula)

### PR DESCRIPTION
**Pull request by kiraum**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cody.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  homepage "https://sourcegraph.com"
+  url "https://registry.npmjs.org/@sourcegraph/cody/-/cody-5.5.15.tgz"
+  sha256 "2f25357f26e7710b71e4abead3ad9ee3fc5817fbbd6a78cd6726556982bc935a"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    ENV["NPM_CONFIG_PREFIX"] = libexec.to_s
+    system "npm", "install", *std_npm_args, "@sourcegraph/cody"
```
Comment (SMillerDev) [inline Formula/c/cody.rb:12]: This will just install any version right? It won't install from the source that's downloaded.

File: Formula/c/cody.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    (bin/"cody").atomic_write <<~EOS
+      #!/bin/bash
+      export NODE_NO_WARNINGS=1
+      exec "#{libexec}/bin/cody" "$@"
+    EOS
+  end
+
+  test do
+    system bin/"cody", "--version"
```
Comment (SMillerDev) [inline Formula/c/cody.rb:24]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

File: Formula/c/cody.rb (hunk 1 above)
Comment (kiraum) [reply Formula/c/cody.rb]: Thanks @SMillerDev  for the review; version now matches source code for better control.

File: Formula/c/cody.rb (hunk 2 above)
Comment (kiraum) [reply Formula/c/cody.rb]: Thanks @SMillerDev, replaced basic version test with assert test for failed auth due to required credentials.
Please, let me know if I need that I improve something else.

**Comment by kiraum** [comment]
I have some tests failing, I will check and ping you asap they are fixed.

**Comment by kiraum** [comment]
@SMillerDev , sorry for the delay,  this is my first public formula, could you please review it when you have a moment?

Thanks!

File: Formula/c/cody.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  homepage "https://sourcegraph.com"
+  url "https://registry.npmjs.org/@sourcegraph/cody/-/cody-5.5.15.tgz"
+  sha256 "2f25357f26e7710b71e4abead3ad9ee3fc5817fbbd6a78cd6726556982bc935a"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    ENV["NPM_CONFIG_PREFIX"] = libexec.to_s
+    system "npm", "install", *std_npm_args, "@sourcegraph/cody@5.5.15"
```
Comment (SMillerDev) [inline Formula/c/cody.rb:12]: From my knowledge this will still install from npm, not from the local source. Looking at `bower` for example, it uses:
```suggestion
    system "npm", "install", *std_npm_args
```

Comment (kiraum) [reply Formula/c/cody.rb]: Hi @SMillerDev , tested your suggestion and works like a charm! Thanks again! 😀 
Can you please check when you have a moment?

**Comment by kiraum** [comment]
@SMillerDev , if/when you have some time, can you take a look?

File: Formula/c/cody.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    ENV["NPM_CONFIG_PREFIX"] = libexec.to_s
+    system "npm", "install", *std_npm_args
+    rm Dir["#{bin}/*"]
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+
+    (bin/"cody").atomic_write <<~EOS
+      #!/bin/bash
+      export NODE_NO_WARNINGS=1
+      exec "#{libexec}/bin/cody" "$@"
+    EOS
```
Comment (SMillerDev) [inline Formula/c/cody.rb:20]: Why is this needed? Does it not create its own binaries?

Comment (kiraum) [reply Formula/c/cody.rb:20]: To avoid node warning to users using Cody cli:
```
» node cody-agent/package/dist/index.js chat -m "hey"
`⠋ Logging in(node:21993) [DEP0040] DeprecationWarning: The punycode module is deprecated. Please use a userland alternative instead.
(Use node --trace-deprecation ... to show where the warning was created)`
Hello! I'm Cody, an AI coding assistant from Sourcegraph. How can I help you with coding or development tasks today? Whether you need help with writing code, debugging, explaining concepts, or anything else related to programming, I'm here to assist. What would you like to work on?

✔ Claude 3.5 Sonnet (29 tokens/second)

```

Comment (SMillerDev) [reply Formula/c/cody.rb:20]: That really just sounds like 
> it works without patching on all Homebrew-supported OS versions

https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff

---

I'd say we wait for the next version or we drop the wrapper because in my experience "I'll fix the technical debt later" rarely comes to pass for formula

Comment (kiraum) [reply Formula/c/cody.rb]: Fair enough, removed the wrapper.

File: Formula/c/cody.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  sha256 "2f25357f26e7710b71e4abead3ad9ee3fc5817fbbd6a78cd6726556982bc935a"
+  license "MIT"
+
+  depends_on "node"
+
+  def install
+    ENV["NPM_CONFIG_PREFIX"] = libexec.to_s
+    system "npm", "install", *std_npm_args
+    rm Dir["#{bin}/*"]
+    bin.install_symlink Dir["#{libexec}/bin/*"]
```
[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb:14]: Based on your experience, do you think that's better to remove it?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/c/cody.rb:14]: I'm just uncertain of the benefits. All Homebrew formulae are build in a temp directory anyway, so there would never be any leftovers. I think it's better to give it a try without this.

[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb:14]: Good to known! Removing `rm Dir["#{bin}/*"]` (just committed removing it), there was not difference when installing, uninstalling, reinstalling between same/different versions, but without `bin.install_symlink Dir["#{libexec}/bin/*"]`,  the binary is available (need to fix PATH manually), so the symlink we need to keep.

[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb]: @SMillerDev , please let me know if more comments or if I should resolve this thread.

[the remark this answers is not in the document]
Comment (SMillerDev) [reply Formula/c/cody.rb]: That is probably an effect of setting the install location to `libexec`, does it work if you just keep the default install location?

[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb]: @SMillerDev Just keeping default local, doesn't work out of the box, since install path doesn't exist natively in the PATH env variable (in truth, it works, but the binary is hidden in the install path, so for some of the users will not "work"), so the symlink still required, but I was able to simplify a little bit, can you please take a look when possible?

**Comment by kiraum** [comment]
@SMillerDev , Is this approach viable for moving forward? If not, no worries - I can continue using my personal tap, close this PR, and revisit in the future if Cody CLI aligns better with the project requirements.
The main goal here was to simplify things for other users.

File: Formula/c/cody.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb]: @SMillerDev , Is this approach viable for moving forward? If not, no worries - I can continue using my personal tap, close this PR, and revisit in the future if Cody CLI aligns better with the project requirements.
The main goal here was to simplify things for other users.

[the remark this answers is not in the document]
Comment (kiraum) [reply Formula/c/cody.rb]: I'm resolving this thread, because there is no answers for a long time, and if not able to move forward, I will close the PR.
